File: src/main/scala/org/oddjob/dido/DataException.scala
package org.oddjob.dido

/**
 * Something has gone wrong with reading or writing data.
 * 
 * @author rob
 *
 */
class DataException(message: String, cause: Throwable)
		extends Exception(message, cause) {
	private val serialVersionUID: Long = 2010072700L

	def this(message: String) = {
		this(message, null)
	}

	def this(cause: Throwable) {
		this(null, cause)
	}

}

File: dido-pickles/src/main/java/org/oddjob/dido/DataException.java
package org.oddjob.dido;

/**
 * Something has gone wrong with reading or writing data.
 * 
 * @author rob
 *
 */
public class DataException extends Exception {
	private static final long serialVersionUID = 2010072700L;
	
	public DataException() {
	}

	public DataException(String message) {
		super(message);
	}

	public DataException(Throwable cause) {
		super(cause);
	}

	public DataException(String message, Throwable cause) {
		super(message, cause);
	}

}
